backends/p4tools/modules/testgen/targets/pna/test/PNAMetadataXfail.cmake

```
@@ -4,43 +4,43 @@
 
 p4tools_add_xfail_reason(
   "testgen-p4c-pna-metadata"
-  "Unknown or unimplemented extern method: recirculate"
+  "Unknown or unimplemented extern method: (.)*.recirculate"
```

I'm not sure what is going on here with the brackets, but this would match things like "[...] method: _recirculate", if you want to match "something" DOT "recirculate" maybe use:

```suggestion
  "Unknown or unimplemented extern method: .*[.]recirculate"

---

Let me try this. I had a bit of trouble finding the right regex pattern for these CMake matches.

---

`(.)*.` matches but `.*[.]` does not. `(.)*` matches 0 or more arbitrary characters and `.` is the dot.

---

yep, it is tricky. Especially since the regex escaping conflicts with CMake escaping, which is why I prefer `[...]` instead of `\` in most cases.